### ISSUE TITLE
Zip - ttf files

### ISSUE THREAD

**Issue by dud3** [Stuk/jszip#277]
Is there a way to zip `ttf` files?

Thanks.

**Comment by dduponchel** [comment]
Yes, as long as you have the files in a binary format.
We will need more information on what you are trying to do (and/or code snippet) and on which platform (browser, nodejs) to give you pertinent examples.

**Comment by dud3** [comment]
Sure, I'll make one quickly.

**Comment by dud3** [comment]
I was trying to achieve something alike the following: https://jsfiddle.net/w253qho0/8/

Note:
```
$.get('https://dud3.bitbucket.org/aseets/fonts/arial.ttf', function(data) {
  fontsFolder.file('arial_test_2.ttf', 'http://dud3.bitbucket.org/aseets/fonts/arial.ttf');
...
```

It doesn't load the file, I guess because of `cross domain access` issues, but you can easily download it from the link above and maybe try it on your local machine, if you don't mind. =)

**Comment by dud3** [comment]
> Comment: https://github.com/Stuk/jszip/issues/277#issuecomment-208041177 - updated.

**Comment by dduponchel** [comment]
Your first try, `fontsFolder.file('arial_test_1.ttf', 'http://dud3.bitbucket.org/aseets/fonts/arial.ttf')`, doesn't work because JSZip doesn't trigger ajax requests.

Your second try is the right way to do it, you just have one issue: `$.get` handles text contents and tries to decode the content from its encoding (usually UTF-8). With a binary file, the decoding step corrupts the content. See [this page](https://stuk.github.io/jszip/documentation/howto/read_zip.html#in-the-browser) for more information.

In short, either use JSZipUtils, an `XMLHttpRequest` with `responseType = 'arraybuffer'`, or any ajax framework that let you download binary content.

**Comment by dud3** [comment]
I see, thanks for the info.

I've actually tried https://stuk.github.io/jszip/documentation/howto/read_zip.html#in-the-browser before, but I got an empty `arraybuffer`, any ideas why that might happen?

**Comment by dduponchel** [comment]
No idea. Could you reproduce the issue in a jsfiddle ?

**Comment by dud3** [comment]
I did try the example one more time and it did actually work, sorry, my bad. =)

Feel free to close this issue.

Thanks.

**Comment by dud3** [comment]
Actually I've closed it, if that's OK with you. =)

**Comment by dduponchel** [comment]
That's OK, thanks !
